mimesis/providers/numbers.py

```diff
@@ -66,6 +66,32 @@ def complexes(self, start_real: float = 0, end_real: float = 1,
                     self.random.uniform(start_imag, end_imag, rounding_imag))
             for _ in range(n)]
 
+    def matrix(self, m: int = 10, n: int = 10, num_type: str = 'floats',
+               **kwargs) -> List[List]:
+        """Generate a m x n matrix with random numbers.
+
+        num_type could be "float", "integers" or "complexes".
+
+        :param m: Number of rows.
+        :param n: Number of columns.
+        :param num_type: Type of numbers to generate.
+        :param **kwargs: Other specific arguments.
+        :return: A matrix of random numbers.
+        """
+        gen_func: Callable
+        if num_type == 'floats':
+            gen_func = self.floats
+        elif num_type == 'integers':
+            gen_func = self.integers
+        elif num_type == 'complexes':
+            gen_func = self.complexes
+        else:
+            raise ValueError(
```

Comment:
No need in this when you use `self. _validate_enum()`

Reply:
@lk-geimfari do you mean something like that?

```
def matrix(num_type: NumType = None):
    gen_func = {
        NumType.FLOATS: self.floats,
        NumType.INTEGERS: self.integers,
        NumType.COMPLEXES: self.complexes
    }
    key = self._validate_enum(item=num_type, enum=NumType)
    return gen_func[key](kwargs)
```

Reply:
Not really. [There](https://github.com/lk-geimfari/mimesis/blob/master/mimesis/providers/person.py#L83) is an example.

Reply:
Enums must be defined in `mimesis.enums`.

Reply:
Something like this:

```
_type = self._validate_enum(num_type, NumType)

if not hasattr(self, _type):
    raise AttributeError('Number has not such provider')

return getattr(self, _type)(**kwargs)
```

Reply:
@lk-geimfari nice! I forgot about the `getattr` function!

Reply:
@lk-geimfari please check out my last commit.

I think that 

    if not hasattr(self, nm_type):
        raise AttributeError('Number has not such method')

is not needed because the argument is already checked in `_validate_enum`